src/wayland/base/wayland_base.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright © 2019 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored By: William Wold <william.wold@canonical.com>
+ */
+
+#ifndef MIR_WAYLAND_OBJECT_H_
+#define MIR_WAYLAND_OBJECT_H_
+
+struct wl_resource;
+struct wl_global;
+struct wl_client;
+
+namespace mir
+{
+namespace wayland
+{
+
+class Resource
+{
+public:
+    Resource();
+    virtual ~Resource() = default;
```

Comment:
For consistency I think we should `delete` CopyAssign semantics.

Reply:
Done